Update pt-mysql-summary

MASTERGOKUL opened the pull request:
Changed the deprecated variable name tx_isolation to transaction_isolation

"`transaction_isolation` was added in MySQL 5.7.20 as an alias for `tx_isolation`, which is now deprecated and is removed in MySQL 8.0. Applications should be adjusted to use `transaction_isolation` in preference to `tx_isolation`"

reference : https://dev.mysql.com/doc/refman/5.7/en/server-system-variables.html#sysvar_tx_isolation

MASTERGOKUL committed:
Update pt-mysql-summary

Changed the deprecated variable name tx_isolation to transaction_isolation

it-percona-cla commented:
[![CLA assistant check](https://cla-assistant.percona.com:443/pull/badge/signed)](https://cla-assistant.percona.com:443/percona/percona-toolkit?pullRequest=839) <br/>All committers have signed the CLA.

svetasmirnova commented:
Thank you for your contribution. You changed the auto-generated code. The fix should go into the lib/bash/report_mysql_info.sh file. Please fix, then run `./util/update-modules bin/pt-mysql-summary `, then run regression tests (I believe they also need to be adjusted).

Also, pt-mysql-summary should support older versions, so we need a condition here. Also, tests should be adjusted for 5.7.20+ and elder versions.

MASTERGOKUL committed:
Merge branch 'percona:3.x' into patch-1

MASTERGOKUL committed:
added version check for transaction_isolation and tx_isolation

MASTERGOKUL committed:
updated transaction_isolation check using ./util/update-modules

MASTERGOKUL commented:
hello @svetasmirnova  I updated the code with mysql version checks it is working fine as I run it. I am not aware of the regression tests, could you please infer me how to do it.

![image](https://github.com/percona/percona-toolkit/assets/94166506/49fdd391-c11f-4ef7-b455-4fd038ce6dec)
working fine actually 👆

svetasmirnova commented on lib/bash/report_mysql_info.sh:
```diff
@@ -1014,6 +1014,18 @@ section_innodb () {
+   # dynamically allocate variable name - transaction_isolation
+   local mysql_version=$(get_var version "$variables_file" | awk -F'-' '{print $1}')
+   local transaction_isolation_var="tx_isolation"
+   version_greater_equal() { # Function to compare versions
```
Why don't you compare version strings like we do on line 229 of `lib/bash/collect_mysql_info.sh`:
```if [ "$version" '<' "8.0" ]; then```

MASTERGOKUL replied:
i thought that it could be good if we specify exact version like 5.7.20,  due to it is not a number we cant able to use `-ge`, is it ok if i use it like `if ["$mysql_version" '>' "5.7.19"] then`

MASTERGOKUL committed:
changed if condition more readable

MASTERGOKUL committed:
Update Changelog typo mistake

svetasmirnova committed:
PR-839 -  Update pt-mysql-summary

- Reversed condition, so default is transaction_isolation, not
  tx_isolation
- Removed extra variable: we have variable $version, defined before
- Fixed test cases

svetasmirnova commented:
Thank you for the changes. I added a few more. Please review them and let us know if you agree.

svetasmirnova committed:
Merge branch '3.x' into HEAD

MASTERGOKUL commented:
> Thank you for the changes. I added a few more. Please review them and let us know if you agree.

I agree with the changes, and thank you for adding it.
